### ISSUE TITLE
New submissions - update searchindex failure

### ISSUE THREAD

**Issue by StuartMcD** [solspace/craft3-freeform#1284]
### What happened?

Front end form. Fill it out and submit. All good, all data recorded, there's an associated Craft entry integration that's being suppressed on initial form submission.

Jump in backend, and Queue manager reports 'Updating search indexes - failed'

### Errors and Stack Trace (if available)

```shell
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '40000-userid-0-1' for key 'PRIMARY' The SQL being executed was: INSERT INTO `searchindex` (`elementId`, `attribute`, `fieldId`, `siteId`, `keywords`) VALUES (40000, 'userid', 0, 1, ' 17732 ') 

Job data:
{
    "description": null,
    "elementType": "Solspace\\Freeform\\Elements\\Submission",
    "elementId": 40000,
    "siteId": 1,
    "fieldHandles": []
}
```

### How can we reproduce this?

1. Create form w/integration with Craft entries.
2. Wire up form to record all info AND suppress integration from triggering
3. Submission successfully recorded, email sent to user
4. Jump in backend, error reported.

### Freeform Edition

Pro

### Freeform Version

5.1.19.1

### Craft Version

4.9.2

### When did this issue start?

Unsure

### Previous Freeform Version

5.1.18

**Comment by padmedia** [comment]
I'm also getting the same error

1. Craft CMS 4.9.4
2. Freeform Pro: 5.2.0
 
```
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '370386-iberiainformationdeclarat-0-1' for key 'searchindex.PRIMARY' The SQL being executed was: INSERT INTO `searchindex` (`elementId`, `attribute`, `fieldId`, `siteId`, `keywords`) VALUES (370386, 'iberiainformationdeclarat', 0, 1, '') 

{
    "description": null,
    "elementType": "Solspace\\Freeform\\Elements\\Submission",
    "elementId": 370386,
    "siteId": 1,
    "fieldHandles": []
}
```

**Comment by kjmartens** [comment]
Hi @StuartMcD and @padmedia,

I'm very sorry for the trouble you're experiencing. We are investigating what might be causing this issue.

Curious... is the error always for the same field? Does it happen for ALL forms or just one/some? And can you tell me more about the affected forms (e.g. map submission data to another Element, attach/relate submissions to another element, etc)?

@StuartMcD I tried reproducing with your steps provided, but I can't trigger the issue for some reason. Just to be clear, when you say `suppress integration from triggering`, you mean this?

``` twig
{{ form.render({
    disable: {
        elements: true
    }
}) }}
```

Thanks! 🙂

**Comment by padmedia** [comment]
Hey Kelsey, thanks for looking into this 😊

It only happens for one field, which is only used on one form. However, submitting to any form will trigger the failed Updating search indexes.  

I must admit I've not looked more deeply into how the searchindex.PRIMARY is generated (in my case an example duplicate error key is: 370386-iberiainformationdeclarat-0-1).

What seems to be very interesting is I have two checkbox fields with similar handles used on two different forms (field handle below):

1. IberiaInformationDeclaration
2. iberiaInformationDeclaration_1

If I change e.g. (iberiaInformationDeclaration_1) to (1iberiaInformationDeclaration) this now resolves the issue.

I wonder if the issue occurs because the field handle is truncated or manipulated in the primary key, which presents as a duplicate? 

@StuartMcD - Do you have two similar field handles by any chance?

I hope that helps?

Thanks ☺️

**Comment by StuartMcD** [comment]
I wanna say yes but I'm running free on this pet project, like proper finger snapping changes on a whim. 

I renamed a field for an existing form/submission, I've then created a new form with different stuff but both have elements generated and then yet I turn an element creation option off and leave another one on and errors still occur.

I truly wish it could be something so mundane and if I was a betting man, I'd look at 'elements' - then again I'm a designer who can code and not the inverse.

**Comment by darylknight** [comment]
I'm also running into this problem. I recently inherited a Craft 2 site, which I've updated to Craft 4 / Freeform 5. Every time someone submits a form, I get this:

![image](https://github.com/solspace/craft-freeform/assets/1250017/bd674038-4e6d-4ae5-99f9-f386fa4945d3)

**Comment by kjmartens** [comment]
Hi @StuartMcD, @padmedia, @darylknight,

Thank you for your patience on this issue...

We've just released Freeform 5.3.1, which should resolve this issue. Can you test and confirm? 🙂

**Comment by darylknight** [comment]
This seems to have solved the issue, thank you!
